pkg/products/monitoringspec/reconciler.go

```diff
@@ -276,14 +276,56 @@ func (r *Reconciler) reconcileRoleBindingsForServiceMonitor(ctx context.Context,
 	}
 	//Create role binding for each of the namespace label selectors
 	for _, namespace := range sermon.Spec.NamespaceSelector.MatchNames {
-		err := r.reconcileRoleBinding(ctx, serverClient, namespace)
+		err := r.reconcileRole(ctx, serverClient, namespace)
+		if err != nil {
+			return err
+		}
+		err = r.reconcileRoleBinding(ctx, serverClient, namespace)
 		if err != nil {
 			return err
 		}
 	}
 	return err
 }
 
+func (r *Reconciler) reconcileRole(ctx context.Context,
+	serverClient k8sclient.Client, namespace string) (err error) {
+
+	role := &rbac.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleBindingName,
```

Comment:
Given that the variable `roleBindingName` is now used also for a "Role" here (so not just a RoleBinding), I think maybe the variable should be renamed to something like `prometheusK8s` or something.

Reply:
Sorry, I had another constant for rolename, updated it with the correct variable. Fixed.